qermit/zero_noise_extrapolation/zne.py

```diff
@@ -396,7 +398,41 @@ def odd_gate(circ: Circuit, noise_scaling: int, **kwargs) -> Circuit:
         c_dict.update({"commands": folded_command_list})
         folded_c = Circuit.from_dict(c_dict)
 
-        return folded_c
+        return [folded_c]
+
+    @staticmethod
+    def noise_aware(circ: Circuit, noise_scaling: int, **kwargs) -> List[Circuit]:
+        """Scale noise in a circuit by adding noisy gates as defined
+        by the given noise model.
+
+        :param circ: Circuit with noise to be scaled.
+        :type circ: Circuit
+        :param noise_scaling: Factor by which noise should be scaled.
+        :type noise_scaling: int
+        :return: List of circuits with additional noise gates added.
+        :rtype: List[Circuit]
+
+        :key noise_model: Noise model defining noise types and rates.
+            Defaults to noiseless model.
+        :type noise_model: NoiseModel
+        :key n_noisy_circuit_samples: The number of random noisy
+            circuits to generate. Defaults to 1.
+        :type n_noisy_circuit_samples: int
+        """
+
+        noise_model = kwargs.get("noise_model", NoiseModel(noise_model={}))
```

Comment:
I think this is my line where I get too pedantic with types in python, but I'd do: `noise_model: NoiseModel = ...` - understood if you'd prefer not to though!

Reply:
Thanks! I think you're right that annotating kwargs is best.